tests/parser/syntax/test_unbalanced_return.py

```diff
@@ -0,0 +1,106 @@
+import pytest
+from pytest import (
+    raises,
+)
+
+from vyper import (
+    compiler,
+)
+from vyper.exceptions import (
+    StructureException,
+)
+
+fail_list = [
+    """
+@public
+def foo() -> int128:
+    pass
```

Comment:
I feel like `pass` should generate a warning

Reply:
I think this is correct, because it expects an int, and this doesn't return.

Reply:
This may be debateable. I was thinking if someone didn't want to implement any functionality to start. But, if it fails to return anything on the stack that's bad too.

Reply:
Yip, kind of wanted to not scope creep... But will add the specific case for code after an return.

Reply:
@fubuloubu `pass` won't put anything into the return buffer. So on the EVM level, the caller will get junk in the buffer they provided to `CALL`.